code/parachain/frame/composable-traits/src/currency.rs

```diff
@@ -142,13 +160,122 @@ impl<
 {
 }
 
-// hack to imitate type alias until it is in stable
-// named with like implying it is`like` is is necessary to be `AssetId`, but may be not enough (if
-// something is `AssetIdLike` than it is not always asset)
+pub trait AssetIdLike = FullCodec + MaxEncodedLen + Copy + Eq + PartialEq + Debug + TypeInfo;
 
-// FIXME(hussein-aitlahcen): this trait already exists in frame_support, named `AssetId`
-pub trait AssetIdLike:
-	FullCodec + MaxEncodedLen + Copy + Eq + PartialEq + Debug + TypeInfo
-{
+#[derive(
+	RuntimeDebug,
+	Copy,
+	Clone,
+	PartialEq,
+	Eq,
+	PartialOrd,
+	Ord,
+	Encode,
+	Decode,
+	MaxEncodedLen,
+	TypeInfo,
+)]
+pub struct Rational64 {
+	pub n: u64,
+	pub d: u64,
+}
+
+pub trait RationalLike<const N: u64, const D: u64> {
+	fn new() -> Self;
+	const CHECK: bool = if D != 0 { true } else { panic!("denominator cannot be zero") };
+}
```

Comment:
I really don't think this trait is necessary - as far as I can tell it's never used also?

Reply:
In theory it can be fixed using NonZero, but had bad experience trying to use NonZero in Substrate. 
So const check always work well.

Reply:
but yeah, may be replacement with NonZero (assuming it encodes into usual number) and it can be const generic, is nicer

Reply:
improved. for non zero keeped other issue opened.

Reply:
I think using NonZero and having a few simple tests to check it's encoding would be good

Reply:
const CHECK: () = assert!(D != 0, "denominator cannot be zero");

Reply:
Sure, NonZero may work if to try.